components/raftstore/src/store/peer_storage.rs

```diff
@@ -515,70 +545,52 @@ fn validate_states<EK: KvEngine, ER: RaftEngine>(
 ) -> Result<()> {
     let last_index = raft_state.get_last_index();
     let mut commit_index = raft_state.get_hard_state().get_commit();
-    let apply_index = apply_state.get_applied_index();
-
-    if commit_index < apply_state.get_last_commit_index() {
-        return Err(box_err!(
-            "raft state {:?} not match apply state {:?} and can't be recovered.",
-            raft_state,
-            apply_state
-        ));
-    }
     let recorded_commit_index = apply_state.get_commit_index();
+    let state_str = format!(
```

Comment:
How about make it a function for lazy evaluation?

Reply:
Done.